src/domain/tokens/entities/schemas/token.schema.ts

```diff
@@ -5,11 +5,11 @@ export const tokenSchema: JSONSchemaType<Token> = {
   type: 'object',
   properties: {
     address: { type: 'string' },
-    decimals: { type: 'number' },
+    decimals: { type: 'number', nullable: true },
```

Comment:
Good catch. This only applies to `ERC721` but maybe we should split this schema into multiple ones and use a `discriminator` for the `type`.

Something that can be tackled separately. 👍